File: 3191. Minimum Operations to Make Binary Array Elements Equal to One I/Solution.py
from typing import List


class Solution:
    def minOperations(self, nums: List[int]) -> int:
        ans = 0
        for i in range(len(nums) - 2):
            if nums[i] == 0:
                nums[i + 1] ^= 1
                nums[i + 2] ^= 1
                ans += 1
        return ans if nums[-2] and nums[-1] else -1

File: 3191. Minimum Operations to Make Binary Array Elements Equal to One I/Solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int minOperations(vector<int>& nums)
    {
        int ans = 0, n = nums.size();
        for (int i = 0; i < n - 2; ++i) {
            if (nums[i] == 0) {
                nums[i + 1] ^= 1;
                nums[i + 2] ^= 1;
                ++ans;
            }
        }
        return nums[n - 2] && nums[n - 1] ? ans : -1;
    }
};
